### PR TITLE
Generate SAS token for OAuth2 Azure

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java [hunk 1]
```diff
@@ -182,7 +182,7 @@ protected final boolean isHierarchical()
     @Override
     protected boolean supportsPreSignedUri()
     {
-        return azureAuth.supportsPreSignedUri();
+        return true;
```
Comment (losipiuk) [inline lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java:185]: when do you set false?

**Review by losipiuk — APPROVED** [review]

File: lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java (hunk 1 above)
Comment (wendigo) [reply lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java:185]: Actually it's not needed anymore I think. Let me check

Comment (wendigo) [reply lib/trino-filesystem-azure/src/test/java/io/trino/filesystem/azure/AbstractTestAzureFileSystem.java:185]: it's false in the parent class. this is true for s3/gcs/abfs but not for hdfs or local file system

**Comment by mosabua** [comment]
no release notes right?

**Comment by wendigo** [comment]
@mosabua no, it's not used beside the spooling which we don't want to announce yet

**Pull request merged**